subsys/net/ip/ipv4.c

```diff
@@ -315,7 +319,7 @@ enum net_verdict net_ipv4_input(struct net_pkt *pkt)
 
 	if (verdict == NET_DROP) {
 		goto drop;
-	} else if (hdr->proto == IPPROTO_ICMP) {
+	} else if (proto == IPPROTO_ICMP) {
```

Comment:
maybe it would be just simpler to check on the verdict at line 305 directly and goto drop or return relevantly there.

Reply:
As you wish, I have changed it now. Don't know what the preferred style is here.

Reply:
It was just an idea. Either way, it's same in the end.

Reply:
No problem.